src/lib/gateway-base.ts

```diff
@@ -7,13 +15,27 @@ class GatewayBase {
   private readonly endpoint: string;
   private readonly certs: StarkExCertsConfig;
 
-  constructor(config: StarkExClientConfig, path: string) {
+  constructor(
+    config: StarkExClientConfig,
+    private readonly path: ApiGatewayPath
+  ) {
     const {endpoint, certs} = config;
-    this.initLogger(path);
-    this.endpoint = `${endpoint}${path}`;
-    if (certs) {
-      this.certs = certs;
-    }
+    this.endpoint = endpoint;
+    this.certs = certs;
+
+    this.initLogger(this.getEndpoint());
+  }
+
+  private getEndpoint(version?: ApiVersion) {
+    version = version === undefined ? this.path.defaultVersion : version;
+
+    return [
+      this.endpoint,
+      mapApiVersionToUrlPrefix[version],
+      this.path.gatewayRoute
+    ]
+      .filter(a => !!a)
+      .join('/');
   }
```

Comment:
```suggestion
    return [
      this.endpoint,
      mapApiVersionToUrlPrefix[version || this.path.defaultVersion],
      this.path.gatewayRoute
    ]
      .filter(a => !!a)
      .join('/');
  }
```

Reply:
WDYT?
I like it more elegant like this, or you prefer the strict comparison?

Reply:
I like it more as well.